src/App.tsx

```diff
@@ -1,18 +1,138 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Routes,
+  Route,
+  useLocation,
+  Navigate,
+} from 'react-router-dom';
 import { PeoplePage } from './components/PeoplePage';
 import { Navbar } from './components/Navbar';
-
+import { HomePage } from './components/HomePage';
+import { PageNotFound } from './components/PageNotFound';
+import { Person, NewPerson } from './types';
+import { getPeople } from './api';
+import { FilterType } from './types/enum';
+import { Loader } from './components/Loader';
 import './App.scss';
 
-export const App = () => {
+export const App: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [newPeople, setNewPeople] = useState<NewPerson[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [sexFilter, setSexFilter] = useState(FilterType.All);
+  const [query, setQuery] = useState('');
+  const [isError, setIsError] = useState(false);
+  const location = useLocation();
+
+  const isHomePage = location.pathname.endsWith('/home');
+
+  const sexFilterHandler = (value: FilterType) => {
+    setSexFilter(value);
+  };
+
+  const deleteQuery = () => {
+    setQuery('');
+    localStorage.setItem('query', '');
+  };
+
+  const fetchPeople = async () => {
+    try {
+      const fetchedData = await getPeople();
+
+      setLoading(false);
+      setPeople(fetchedData);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('Error occurred:', error);
+      setIsError(true);
+    }
+  };
+
+  useEffect(() => {
+    const savedQuery = localStorage.getItem('query');
+
+    if (savedQuery) {
+      setQuery(savedQuery);
+    }
+
+    fetchPeople();
+  }, []);
+
+  useEffect(() => {
+    const updatedPeople = people.map((person, index) => {
+      return {
+        ...person,
+        index: index + 1,
+      };
+    });
+
+    setNewPeople(updatedPeople);
+  }, [people]);
+
   return (
     <div data-cy="app">
       <Navbar />
 
       <div className="section">
         <div className="container">
-          <h1 className="title">Home Page</h1>
-          <h1 className="title">Page not found</h1>
-          <PeoplePage />
+          <h1 className="title">People Page</h1>
+          <Routes>
+            {isHomePage && (
+              <Route path="home" element={<Navigate to="/" replace />} />
+            )}
+            <Route path="*" element={<PageNotFound />} />
+            <Route path="/" element={<HomePage />} />
+            <Route
+              path="/people"
+              element={
+                loading ? (
+                  <Loader />
+                ) : (
+                  <PeoplePage
```

Comment:
Move this logic to the PeoplePage. You can move all logic for the people table to the people page. You don't need to save all these variables inside the App.tsx.
The App.tsx file is typically used for overall application-level logic and configuration, while the PeoplePage component is more specific to displaying and managing the people table. Separating these concerns makes the codebase more modular and easier to understand. It'll allow you to reuse that component in different parts of your application, without being tightly coupled to the App.tsx file.